### ISSUE TITLE
Add a `useField` hook to be used in custom components

### ISSUE THREAD

**Issue by bvangraafeiland** [SeasonedSoftware/remix-forms#124]
As of now, when adding custom components to the form, there is no easy way to customize them based on the form state. We have the `renderField` prop but that leads to [verbose code](https://remix-forms.seasoned.cc/examples/render-field/inline-checkboxes) to keep the layout for different field types intact.

Part of this is adding a context to allow RHF's `useFormContext` to be used (see #37). This doesn't include the individual field states though (defined in https://github.com/SeasonedSoftware/remix-forms/blob/main/packages/remix-forms/src/createForm.tsx#L47-L62).

A relatively simple solution would be to add something like a `FieldContextProvider` around each field, where the props of the `Field` type are passed through. This way, custom components (input, but also label etc) could easily implement things like error and required indicators, like such:

```
const MyInput = (props) => {
  const { errors } = useField();
  return <input {...props} className={errors ? 'border-red-600' : 'border-gray-200'} />;
}

const MyLabel = ({ children, ...props }) => {
  const { required } = useField();
  return <label {...props}>{children}{required ? ' *' : ''}</label>;
}
```

Basically this will allow the customization that the render props have currently, but take it a step further and make it easier to have customizations applied by default.

**Comment by bvangraafeiland** [comment]
Now that I'm in the flow, I wouldn't mind going for a PR to implement this one as well, if that's ok?

**Comment by danielweinmann** [comment]
Go for it, @bvangraafeiland 💪🏼 

My only suggestion is that we call the provider `FieldProvider` instead of `FieldContextProvider`, to be consistent with `FormProvider`. What do you think?

**Comment by bvangraafeiland** [comment]
Sure thing!

**Comment by bvangraafeiland** [comment]
Actually ended up not exposing the context provider at all, so there was no need for an intermediate component. The context provider is applied directly on the field component now.

**Comment by danielweinmann** [comment]
[1.4.0](https://github.com/SeasonedSoftware/remix-forms/releases/tag/v1.4.0) released it 🎉 

Thank you, @bvangraafeiland, for the great contribution!
